addons/mass_mailing/tests/test_mailing_internals.py

```diff
@@ -313,6 +313,62 @@ def test_mailing_computed_fields_default(self):
         })
         self.assertEqual(literal_eval(mailing.mailing_domain), [('email', 'ilike', 'test.example.com')])
 
+    @users('user_marketing')
+    def test_mailing_computed_fields_default_email_from(self):
+        # Testing if the email_from is correctly computed when an
+        # alias domain for the company is set
+
+        # Case where alias domain is set and no outgoing email server for mass_mailing
+        mailing = self.env['mailing.mailing'].create({
+            'name': 'TestMailing no default mail server for mass mailing',
+            'subject': 'Test',
+        })
+        self.assertEqual(mailing.email_from, self.env.user.email_formatted)
+
+        # Case where alias domain is set and there is a default outgoing email server
+        # for mass mailing. from_filter matches domain of company alias domain
+        # before record creation
+        mass_mailing_mail_server = self.env['ir.mail_server'].sudo().create({
+            'name' : 'mass_mailing_test',
+            'from_filter' : 'test.mycompany.com',
+            'smtp_host' : 'not_real@smtp.com'
+        })
+
+        self.env['res.config.settings'].sudo().create({
+            'mass_mailing_mail_server_id' : mass_mailing_mail_server.id,
+            'mass_mailing_outgoing_mail_server' : True
+        }).execute()
+
+        mailing_alias_domain = self.env['mailing.mailing'].create({
+            'name': 'TestMailingAliasDomain matches',
+            'subject': 'Test',
+        })
+        self.assertEqual(mailing_alias_domain.email_from, self.env.user.company_id.alias_domain_id.default_from_email)
+
+        # Case where alias domain is set and there is a default outgoing email server
```

Comment:
Actually could be done with a small loop + subTest, to test our various use cases without having to repeat the code too much

Reply:
Not sure what you meant with a loop. Tried to meet you half-way with a helper function as to remove repetitive code. Not sure if that is what you wished for. Always happy to learn/understand any other approach you might be seeing as more efficient or up to standard

Reply:
Ah yes, it might not be clear if you are not used to it. Idea is to use subTest, with input / expected output, like
```
sources = [values1, values, 2]
expected = [expected1, expected 2]
for source, expected in zip(source, expected):
    with self.subTest(source=source):  # used notably when failing to ease debug
        self.assertEqual(do_stuff(source), expected)
```
You can have a look at existing tests using subTest notably in test_mail addon. It allows to have a clear input / expected declaration, then the test itself in the loop, and there is few code redundancy. For example here it could be something like
```
servers = [self.env['ir.mail_server'], from_filter_match, from_filter_missmatch]
expected_from_all = [
    self.env.user.email_formatted,  # default when no server
    self.env.user.company_id.alias_domain_id.default_from_email,  # matches company alias domain
    self.env.user.email_formatted,  # not matching from filter -> back to user from
]
for server, expected_from in zip(servers, expected_from_all):
    with self.subTest(server_name=server.name):
        # create mailing
        ....
        # check email_from
        ....
```

Reply:
Hey @tde-banana-odoo  (and @reth-odoo), thx for taking the time to explain ;)
Learned new things.
So gave it a fair try and pushed a commit. Let me know if I got the gist of it or if there is stuff that can be done in a nice way.